server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/GetSnapshotsResponse.java

```diff
@@ -171,59 +96,56 @@ public boolean isFailed() {
     }
 
     @Override
-    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject();
-        builder.startArray("responses");
-
-        for (Map.Entry<String, List<SnapshotInfo>> snapshots : successfulResponses.entrySet()) {
-            builder.startObject();
-            builder.field("repository", snapshots.getKey());
-            builder.startArray("snapshots");
-            for (SnapshotInfo snapshot : snapshots.getValue()) {
-                snapshot.toXContent(builder, params);
+    public void writeTo(StreamOutput out) throws IOException {
+        if (out.getVersion().onOrAfter(GetSnapshotsRequest.MULTIPLE_REPOSITORIES_SUPPORT_ADDED)) {
+            out.writeMap(failedResponses, StreamOutput::writeString, StreamOutput::writeException);
+        } else {
+            if (failedResponses.isEmpty() == false) {
+                throw failedResponses.values().iterator().next();
```

Comment:
Hmm now that you mention it, I think this is actually dead code now with the changes I made. We would always just straight throw the exception in the transport action when communicating with an old version node and never get here because the old version node would only request snapshots for a single repo. Let me see what I can do about this :)

Reply:
I added an assertion for this in https://github.com/elastic/elasticsearch/pull/74451/commits/fc027a47ef0518301e2a4801989755b0b2d390fd#diff-0f20d081deb6b8f13072b3c96acaee24ba6877e5ac31c867830ec17d65956facR104 and kept the code as it was since I think it's dead code anyways as far as I can see, but just to be safe in case I'm missing something I kept the existing logic around.